src/kvstore/kvstore.cc

```diff
@@ -26,8 +26,10 @@
 #include <stdlib.h>
 #include <dmlc/logging.h>
 #include "./kvstore_local.h"
+
 #if MXNET_USE_DIST_KVSTORE
 #include "./kvstore_dist.h"
+std::atomic<int> mxnet::kvstore::KVStoreDist::customer_id_{0};
```

Comment:
Is there really only one global?  Who/what uses this?
DOes this value every change after it is set to nonzero for the first time?  When?

Reply:
yes, it is supposed to be singleton, 

it is used when creating a new worker https://github.com/apache/incubator-mxnet/pull/9389/files#diff-0cd6fcb2cd941d4c4a829bb3d7ea3d63R54

Reply:
What has changed that requires this now?

Reply:
before, we hardcoded that the customer_id is always 0 since we have only one KVWorker in a process, now, we need to have different customer_id s for different workers, 

and since workers can be created in different threads, we would need to make it thread-safe